test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -103,7 +115,8 @@ func NewMetricsGrabber(c clientset.Interface, ec clientset.Interface, kubelets b
 		grabFromKubelets:          kubelets,
 		grabFromScheduler:         scheduler,
 		grabFromClusterAutoscaler: clusterAutoscaler,
-		masterName:                masterName,
+		kubeScheduler:             kubeScheduler,
+		kubeControllerManager:     kubeControllerManager,
 		registeredMaster:          registeredMaster,
```

Comment:
maybe we should remove the registeredMaster variable and change HasRegisteredMaster too?

```
// HasControllerManagerAndScheduler returns if metrics grabber was able to find the kube-controller-manager and kube-scheduler
func (g *Grabber) HasControllerManagerAndScheduler () bool {
	return kubeScheduler != "" && kubeControllerManager != ""
}
```

the usage is here:
https://github.com/kubernetes/kubernetes/search?q=HasRegisteredMaster&unscoped_q=HasRegisteredMaster

Reply:
Thanks again.
I tend to prefer HasControlPlanePods().
I will update this PR.